test.ps1

```diff
@@ -27,45 +27,56 @@ $tests = @(
     @{ rule = "schema-object-must-have-a-type"; expectError = $false; filename = "schema-object-must-have-a-type-valid.yaml" },
     @{ rule = "schema-object-must-have-a-type"; expectError = $true; filename = "schema-object-must-have-a-type-invalid.yaml" },
     @{ rule = "schema-name-length-must-be-short"; expectError = $false; filename = "schema-name-length-must-be-short-valid.yaml" },
-    @{ rule = "schema-name-length-must-be-short"; expectError = $true; filename = "schema-name-length-must-be-short-invalid.yaml" }
+    @{ rule = "schema-name-length-must-be-short"; expectError = $true; filename = "schema-name-length-must-be-short-invalid.yaml" },
+    @{ rule = "must-use-supported-security-scheme"; expectError = $false; filename = "must-use-supported-security-scheme-valid.yaml" },
+    @{ rule = "must-use-supported-security-scheme"; expectError = $true; filename = "must-use-supported-security-scheme-invalid.yaml" },
+    @{ rule = "must-not-use-basic-auth"; expectError = $false; filename = "must-not-use-basic-auth-valid.yaml" },
+    @{ rule = "must-not-use-basic-auth"; expectError = $true; filename = "must-not-use-basic-auth-invalid.yaml" }
 )
 
-$fileCount = Get-ChildItem (Join-Path $PSScriptRoot "TestSpecs") | Measure-Object | Select-Object -ExpandProperty Count
-if ($tests.Count -ne $fileCount) {
-    throw "Number of tests does not match number of specs. Add the missing specs to the `$tests variable. Files: $fileCount, Tests: $($tests.Count)"
-}
-
-# We expect spectral to fail, so we need to capture the error
-$PSNativeCommandUseErrorActionPreference = $false
-foreach ($test in $tests) {
-    $expectedError = $test.expectError
-    $rule = $test.rule
-    $spec = Join-Path $PSScriptRoot "TestSpecs" $test.filename
+function RunSpectralTests($ruleset, $tests, $testSpecsPath)
+{
+    $PSNativeCommandUseErrorActionPreference = $false
```

Comment:
Not sure if this can be put in a method, I always saw it at the top of the script